### ISSUE TITLE
[BUG] Fix missing key presses and on click events

### ISSUE THREAD

**Issue by kellis-sl** [ScottLogic/sl-tech-carbon-estimator#295]
**Describe the bug**
The ESLint accessibility plugin is highlighting that we are missing these functions in several of the html files and therefore this should be addressed.

**To Reproduce**
1. Add ESLint accessibility plugin to the ESLint config
2. Run ESLint on .html files

**Expected behavior**
No accessbility errors

Example from the export-modal.component.html:
<img width="1039" height="228" alt="Image" src="https://github.com/user-attachments/assets/b9a3d73f-e2e2-4396-a46c-03a2384666d7" />
